src/arctic3d/modules/clustering.py

```diff
@@ -127,11 +127,12 @@ def get_clustering_dict(clusters, ligands):
     cl_dict = {}
     # loop over clusters
     for cl in range(len(clusters)):
-        if clusters[cl] not in cl_dict.keys():
-            cl_dict[clusters[cl]] = [ligands[cl]]
+        if (strcl := str(clusters[cl])) not in cl_dict.keys():
+            cl_dict[strcl] = [ligands[cl]]
```

Comment:
This will change the type of the key from integer to string and can have breaking consequences, why not use integers?

Reply:
This is the point of the PR, integer as keys cannot be loaded using the `json` library.

Reply:
Yes this is the default behaviour of json.load, casting the type into it needs you to reclass a Decoder